crates/utils/src/utils.rs

```diff
@@ -114,6 +115,10 @@ pub fn is_valid_preferred_username(preferred_username: &str) -> bool {
     && preferred_username.chars().count() <= 20
 }
 
+pub fn is_valid_matrix_id(matrix_id: &str) -> bool {
+  VALID_MATRIX_ID_REGEX.is_match(matrix_id)
+}
```

Comment:
We could also check that the domain resolves via DNS, and even get the account info via Matrix API, but thats probably too much for now.

Reply:
Ya I think for things like email, matrix, etc, regex validation is enough. Who knows what networks lemmy will run on.